iris-GmbH/meta-iris-base
# Feature/migl/rdphoen 1164 locate rootfs

Dynamically select the proper rootfs from the kernel cmd line argument firmware_a or firmware_b.

**How to test?**
Check the bootlog which rootfs is started:
```
u-boot=> editenv mmcargs
edit: setenv bootargs ${jh_clk} console=${console} root=${mmcroot} firmware_b
u-boot=> saveenv
u-boot=> boot
[..]
[    3.523320] Run /init as init process
Mount pseudo fs
[    3.550558] random: vgchange: uninitialized urandom read (4 bytes read)
  8 logical volume(s) in volume group "irma6lvm" now active
[    3.692913] random: vgmknodes: uninitialized urandom read (4 bytes read)
Initramfs Bootstrap...
Kernel cmdline: console=ttymxc1,115200 root=/dev/mmcblk2p-22 rootwait rw firmware_b    <--- !!!
select firmware_b
Root mnt   : /mnt
Root device: /dev/mapper/irma6lvm-rootfs_b    <--- !!!
Verity device: /dev/mapper/verity-rootfs_b    <--- !!!
[    3.877222] device-mapper: verity: sha256 using implementation "sha256-caam"
[    3.929295] EXT4-fs (dm-8): mounted filesystem with ordered data mode. Opts: (null)
Switch to root
[..]
IRIS IRMA6 2.0.5-dev imx8mp-irma6r2 /dev/ttymxc1
imx8mp-irma6r2 login:
```

And accordingly repeated for firmware_a:

```
[..]
[    3.519356] Run /init as init process
Mount pseudo fs
[    3.548269] random: vgchange: uninitialized urandom read (4 bytes read)
  8 logical volume(s) in volume group "irma6lvm" now active
[    3.681052] random: vgmknodes: uninitialized urandom read (4 bytes read)
Initramfs Bootstrap...
Kernel cmdline: console=ttymxc1,115200 root=/dev/mmcblk2p-22 rootwait rw firmware_a    <--- !!!
select firmware_a
Root mnt   : /mnt
Root device: /dev/mapper/irma6lvm-rootfs_a    <--- !!!
Verity device: /dev/mapper/verity-rootfs_a    <--- !!!
[    3.877347] device-mapper: verity: sha256 using implementation "sha256-caam"
[    3.927340] EXT4-fs (dm-8): mounted filesystem with ordered data mode. Opts: (null)
Switch to root
[...]
IRIS IRMA6 2.0.5-dev imx8mp-irma6r2 /dev/ttymxc1
imx8mp-irma6r2 login: 
```
